services/beeper-connector/beeper_to_rdf.py

```diff
@@ -0,0 +1,183 @@
+        result = cursor.fetchone()
+        if result and result[0]:
+            user_data = json.loads(result[0])
+            name = user_data.get('fullName', user_id)
+            return name
+        return user_id
+    except:
+        return user_id
+
+def get_thread_info(cursor, thread_id):
+    """Get thread information from the database."""
+    try:
+        cursor.execute("SELECT json_extract(thread, '$.title') FROM threads WHERE threadID = ?", (thread_id,))
+        result = cursor.fetchone()
+        if result and result[0]:
+            return result[0]
+        return thread_id
+    except:
+        return thread_id
+
+def extract_messages_to_rdf(db_path, output_file, limit=10000):
+    """Extract messages from Beeper database and convert to RDF format."""
+    try:
+        conn = sqlite3.connect(db_path)
+        cursor = conn.cursor()
+
+        print(f"Extracting up to {limit} messages from Beeper database...")
+
+        # Get messages with text content from the database
+        cursor.execute("""
+            SELECT
+                roomID,
+                senderContactID,
+                json_extract(message, '$.text') as message_text,
+                timestamp,
+                eventID
+            FROM mx_room_messages
+            WHERE type = 'TEXT'
+            AND json_extract(message, '$.text') IS NOT NULL
+            AND json_extract(message, '$.text') != ''
+            ORDER BY timestamp DESC
+            LIMIT ?
+        """, (limit,))
+
+        messages = cursor.fetchall()
+        print(f"Found {len(messages)} messages with text content.")
+
+        with open(output_file, 'w', encoding='utf-8') as f:
+            # Write RDF header
+            f.write('@prefix : <http://example.org/beeper/> .\n')
+            f.write('@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .\n')
+            f.write('@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .\n')
+            f.write('@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .\n')
+            f.write('@prefix dc: <http://purl.org/dc/elements/1.1/> .\n\n')
+
+            # Process each message and write RDF triples
+            for i, (room_id, sender_id, text, timestamp, event_id) in enumerate(messages):
+                if not text:
+                    continue
+
+                # Process room ID
+                room_name = get_thread_info(cursor, room_id)
+                room_id_safe = re.sub(r'[^a-zA-Z0-9_]', '_', room_id)
+
+                # Process sender ID
+                sender_name = get_user_info(cursor, sender_id)
+                sender_id_safe = re.sub(r'[^a-zA-Z0-9_]', '_', sender_id)
+
+                # Create a safe event ID
+                event_id_safe = re.sub(r'[^a-zA-Z0-9_]', '_', event_id)
+
+                # Format timestamp
+                timestamp_str = datetime.fromtimestamp(timestamp/1000).isoformat()
+
+                # Generate RDF triples
```

Comment:
Ideally, we'd have another helper script which is able to take such "RDF" file (headers + triples) and push it to an evault using our evault/adapter protocols. It is not much more useful to have it in a file than to just get it as json :)